===================================================================
RCS file: /project/cl-store/cvsroot/cl-store/sbcl/custom.lisp,v
retrieving revision 1.8
retrieving revision 1.10
diff -u -r1.8 -r1.10
--- cl-store/sbcl/custom.lisp	2005/05/05 12:58:57	1.8
+++ cl-store/sbcl/custom.lisp	2005/10/04 08:14:02	1.10
@@ -52,9 +52,10 @@
   (slot-value dd 'sb-kernel::name))
 
 (defvar *sbcl-struct-inherits*
-  (list (get-layout (find-class t))
-        (get-layout (find-class 'sb-kernel:instance))
-        (get-layout (find-class 'cl:structure-object))))
+  `(,(get-layout (find-class t))
+    ,@(when-let (class (find-class 'sb-kernel:instance nil))
+        (list (get-layout class)))
+    ,(get-layout (find-class 'cl:structure-object))))
 
 (defstruct (struct-def (:conc-name sdef-))
   (supers (required-arg :supers) :type list)
